src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -1106,7 +1106,9 @@ public override string ToString()
         /// </summary>
         public uint256[] ToHashList()
         {
-            var hashes = new uint256[this.DownloadTo.Height - this.DownloadFrom.Height];
+            int itemsCount = this.DownloadTo.Height == this.DownloadFrom.Height ? 1 : this.DownloadTo.Height - this.DownloadFrom.Height + 1;
```

Comment:
Why the ternary operator here? If the guys are equal, the result will be one

Reply:
This one still holds

Reply:
hmm yeah I will fix